### PR TITLE
[FIX] Rename Protocol classes

### LINKED ISSUE #33: Proposal: rename GraphProtocol to Graph
Hi @adam2392,

I recommend renaming

https://github.com/py-why/dodiscover/blob/3c4f8d251dcf080c4bce83292affff6cd46f335d/dodiscover/_protocol.py#L6

to just `Graph`. In Java it became kind of an anti-pattern to name an entity `ISomeEntity` for interfaces and I would apply the same here. The reason is that it gives you less flexibility in your design to change from a concrete class to a protocol or back, because you'd have to change the name too and everywhere where it's used.

### PR DESCRIPTION
**Pull request by adam2392**
Fixes #33 

Changes proposed in this pull request:

- Updated GraphProtocol -> Protocol
- Updated unit tests to reflect changes in pywhy-graphs

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/dodiscover/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/dodiscover/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

**Comment by bloebp** [comment]
Is there a reason to not be explicit about that they are 'Protocols'?

**Comment by adam2392** [comment]
> Is there a reason to not be explicit about that they are 'Protocols'?

Isn't that what was suggested here tho? https://github.com/py-why/dodiscover/issues/33

**Comment by petergtz** [comment]
> Is there a reason to not be explicit about that they are 'Protocols'?

@bloebp Have a look at https://developer.okta.com/blog/2019/06/25/iinterface-considered-harmful and replace Java interfaces with Python Protocols and the `I`-prefix with the `Protocol`-suffix.

TL;DR: As a user of `Graph`, you don't care that it's a Protocol. You just care that about its methods:

```python
def f(g: Graph):
    g.nodes() ...

f(nx.DiGraph())
```

**Review by petergtz — DISMISSED** [review]
Assuming you'll squash commits before merging.

**Review by petergtz — APPROVED** [review]
